lib/Parse/ParseExpr.cpp

```diff
@@ -525,6 +525,7 @@ ParserResult<Expr> Parser::parseExprUnary(Diag<> Message, bool isExprBasic) {
 ParserResult<Expr> Parser::parseExprKeyPath() {
   // Consume '\'.
   SourceLoc backslashLoc = consumeToken(tok::backslash);
+  llvm::SaveAndRestore<SourceLoc> slashLoc(SwiftKeyPathSlashLoc, backslashLoc);
```

Comment:
This basically to save SwiftKeyPathSlashLoc in the parser context so parseExprPostfixSuffix can use it to create an instance of KeyPathExpr.